### PR TITLE
Fix wsrelay connection in ipv6 environments

### PR DESCRIPTION
**Pull request by kdelee**
##### SUMMARY

Found that in IPv6 only openshift cluster, wsrelay was failing to ever make connections and eventually causing supervisor to exit everything and the pod would exit.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Other

##### AWX VERSION

```
devel
```

##### ADDITIONAL INFORMATION

Working on testing this locally. Using `kind` I've got this config:
```
---
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
name: dev
networking:
  ipFamily: ipv6
nodes:
- role: control-plane
  extraMounts:
  - hostPath: /home/kdelee/projects/ansible/awx
    containerPath: /awx_devel
  extraPortMappings:
  - containerPort: 30080
    hostPort: 30080
```

I have recreated with this, using `VERSION=2.7.1 make deploy` in the `awx-operator` repo to deploy operator and applying:

```
---          
apiVersion: awx.ansible.com/v1beta1
kind: AWX
metadata:
  name: awx
  namespace: awx
spec:
  admin_user: admin
  admin_email: admin@localhost
  service_type: nodeport 
  development_mode: true 
  nodeport_port: 30080 
  image: ghcr.io/ansible/awx_kube_devel
  image_version: devel
```
With my local changes checked out

**NOTE** for some reason I kept getting image pull error on ` ghcr.io/ansible/awx_kube_devel`...I resolved by doing `kind load docker-image ghcr.io/ansible/awx_kube_devel:devel --name dev`

### CODE REVIEW

File: awx/main/wsrelay.py [hunk 1]
```diff
@@ -71,7 +73,16 @@ async def connect(self):
         if not self.channel_layer:
             self.channel_layer = get_channel_layer()
 
-        uri = f"{self.protocol}://{self.remote_host}:{self.remote_port}/websocket/relay/"
+        # figure out if what we have is an ipaddress, IPv6 Addresses must have brackets added for uri
+        uri_hostname = self.remote_host
+        try:
+            # Throws ValueError if self.remote_host is a hostname like example.com, not an IPv4 or IPv6 ip address
+            if type(ipaddress.ip_address(host)) == ipaddress.IPv6Address:
```
Comment (relrod) [inline awx/main/wsrelay.py]: I think `isinstance` is more conventional?

```suggestion
            if isinstance(ipaddress.ip_address(host), ipaddress.IPv6Address):
```

Comment (relrod) [reply awx/main/wsrelay.py]: oh, and:
```suggestion
            if isinstance(ipaddress.ip_address(uri_hostname), ipaddress.IPv6Address):
```

Comment (kdelee) [reply awx/main/wsrelay.py]: good eye, will fix

**Comment by kdelee** [comment]
@relrod pushed suggeted changes

I have successfully recreated the bug with this local kind cluster using devel

Then when I switch to using my code, I have observed the fix working
```
2023-11-03 02:20:32,686 INFO     [-] awx.main.wsrelay Active instance with hostname awx-task-66655b6f64-pxhcb is registered.
2023-11-03 02:20:32,742 WARNING  [-] awx.main.tasks.system Rejoining the cluster as instance awx-task-66655b6f64-pxhcb. Prior last_seen None
2023-11-03 02:20:32,749 WARNING  [-] awx.main.dispatch.periodic Scheduler next run unexpectedly over 20 seconds in future: 25.935721397399902
2023-11-03 02:20:42,695 INFO     [-] awx.main.wsrelay Adding {'awx-web-5fcf6dbc85-rlhpm'} to websocket broadcast list
2023-11-03 02:20:42,713 INFO     [-] awx.main.wsrelay Connection from awx-task-66655b6f64-pxhcb to fd00:10:244::1c established.
2023-11-03 02:20:42,713 INFO     [-] awx.main.wsrelay Starting producer for metrics
```
:tada:

**Review by relrod — APPROVED** [review]

**Pull request merged**